rock/orogen_plugin.rb

```diff
@@ -727,6 +727,32 @@ def update_spec
                         doc("the global name that should be used for the internal #{name} frame")
                 end
             end
+            
+            #also do frame mapping in case the transformations are requested
+            frame_selection = configurable_frames.sort.map do |frame_name|
+                "    _#{name}.setFrameMapping(\"#{frame_name}\", _#{frame_name}_frame);"
+            end.compact
+            #add method to request needed transformations
+            transformationBody = frame_selection.join("\n") + " \n
+    std::vector<base::samples::RigidBodyState > ret;
+    const std::vector<transformer::Transformation *> &transformations(_transformer.getRegisteredTransformations());
+    ret.reserve(transformations.size());
+    
+    for(std::vector<transformer::Transformation *>::const_iterator transform = transformations.begin();
+        transform != transformations.end(); transform++)
+    {
+        base::samples::RigidBodyState curTr;
+        curTr.sourceFrame = (*transform)->getSourceFrame();
+        curTr.targetFrame = (*transform)->getTargetFrame();
+        
+        ret.push_back(curTr);
+    }
+    
+    return ret;"
+
+            task.hidden_operation("getNeededTransformations", transformationBody).
```

Comment:
After applying the patch i get this for several modules:

===== Internal error =======
Please report the following error to the orogen developers
/home/stefan/Software/software_spacebot_test/tools/orogen/lib/orogen/gen/tasks.rb:246:in `register_for_generation': a hidden operation must have a body (OroGen::InternalError)
        from /home/stefan/Software/software_spacebot_test/tools/orogen/lib/orogen/gen/tasks.rb:517:in `each'
        from /home/stefan/Software/software_spacebot_test/tools/orogen/lib/orogen/gen/tasks.rb:517:in `generate'
        from /home/stefan/Software/software_spacebot_test/tools/orogen/lib/orogen/gen/project.rb:675:in `block in generate'
        from /home/stefan/Software/software_spacebot_test/tools/orogen/lib/orogen/gen/project.rb:675:in `each'
        from /home/stefan/Software/software_spacebot_test/tools/orogen/lib/orogen/gen/project.rb:675:in `generate'
        from /home/stefan/Software/software_spacebot_test/tools/orogen/bin/orogen:171:in `<main>'